### ISSUE TITLE
New Maintainer Needed (by end of July)

### ISSUE THREAD

**Issue by rixx** [byro/byro#331]
tl;dr: I'm looking to hand off byro to a group or an individual. If nobody steps up by the end of July, I'll put this repo in "archived" mode to accurately reflect that it's not being maintained or worked on. Please respond here and/or via r at rixx dot de.

-------

So, yeah. For the past five-or-so years, I haven't been running any associations or groups that need something like byro, and have been continuing to vaguely merge patches and updates, but haven't worked on byro itself. That's obviously not ideal, and bit rot is setting in as you'd expect. I have enough other projects and also occasionally a life, so pretending that I'm maintaining byro while doing the bare minimum is clearly the worst solution, as it'll make people believe that the repo is alive, when it's … not.

In an ideal world, one or several of you will decide that you like byro enough to take over maintainership. Talk to me by the end of July, and I can transfer control of the byro org (containing the main repo, plus some plugins). I won't suddenly disappear, so I'm still around for questions, of course!

In a less than ideal world, nobody else has time or interest for byro either. In this case, I'll archive the repo. You'll still be able to access and install it, but there won't be any updates, which will make it eventually unusable (the Django version is already out of support and needs updating).

(I was planning on publishing this on [Volunteer Responsibility Amnesty Day](https://www.volunteeramnestyday.net/), but better late than never). So long, and thanks for all the fish! :rocket:

**Comment by rixx** [comment]
A short word on what a maintainer should be / do / know:

- You're in a group that uses byro.
- You know Python.
- (Optional: you have passing familiarity with Django)
- You'll
    - update dependencies (with the help of dependabot, but sometimes also manual intervention)
    - review PRs
    - respond to issues
    - fix things when they break, like docs not building etc

(That's the minimum, though you can of course share these tasks in a group!)

**Comment by BMaxV** [comment]
Hi,

here from mastodon, I know python and I could do things.

Not part of a group that uses it, but how groups are being run is an interest of mine. It would be cool if there was a way to get in touch with the people who do use it.

I managed to get it working from the docs, so that's a good sign.

So yeah, I'm available, let's see what happens.

**Comment by rennerocha** [comment]
Hello @rixx,

Just discovered byro from your message in Mastodon (I replied to your post). We were looking for a solution to manage membership and payments in our hackerspace (@lhc@mastodon.com.br). We even started a solution like that (with Django), but it didn`t get too far. however, the current state of the project is nice and I loved the UI (one of the things that sometimes is difficult to develop) and its simplicity.

I installed it locally, run a few tests, and apparently it will solve one of our issues (knowing exactly who are our associates and how much are they paying). I am planning to do a proper install in a VPS and add some real data and ask some of the other people from our board to use it a bit. But it is very likely that we will keep using it. Another interesting feature is how plugins can be made. Some integrations with other systems that we have will be possible.

Unfortunately, I won`t be able to use byro a lot, as I will be traveling to Europe next week (for vacation and to attend  EuroPython), but I am very interested and I believe I could keep the project.

**Comment by Cheukting** [comment]
I do not use Byro or Django actively so I am not here to apply for the “job”. However, same as @rennerocha i will be going to EuroPython. If @rixx wants to give this a shout there and if you won’t be there I am happy to help 😊

**Comment by valberg** [comment]
Hi @rixx

I'm a part of an organisation called data.coop in Denmark which has the goal to run various services for its members (mail, mastodon, matrix, etc.) on our own hardware.

We are in the same situation as @rennerocha. We have been trying to get our own system of the ground (https://git.data.coop/data.coop/membersystem) but haven't come around to getting all the gnarly details in order.

I'm going to install byro on our server soonish and give it a spin. From the looks of it it has potential for our organisation.

Long story short; I'm pretty sure that I'm up for helping maintaining (and developing) byro :)

**Comment by rixx** [comment]
Really glad to see so many people show up! If you (@valberg and @rennerocha) end up still being interested after setting it up (sorry for the state of the docs!), please ping me via email and I'll open a conversation with everybody re: how to proceed.

**Comment by Herr-Dante** [comment]
Hallo rixx,

ich hatte dir schon eine Mail geschickt, hast du die bekommen?

Grüße
Dante

**Comment by rixx** [comment]
Ja, ich hatte deine Mail gestern bekommen, aber bin gerade nicht bei taggleichen Antworten (bin nebenher noch mit einer nahenden Großveranstaltung beschäftigt) :wink: 

Wenn du bis zum Wochenende wartest, bekommst du Antwort.

**Comment by karl-baker** [comment]
Hey @rixx 

Sorry for not getting back sooner. Thanks for keeping me in the loop.

During an evaluation of different organizational membership software packages I came across Byro for a small organization that is interested in implementing a package, but like many others has little or no budget and does not want their data to be somewhere in a cloud. After installing Byro in the development environment on a virtual machine on my local system and the production version on hardware of its own I will be presenting the application to the organization in the near future.

What I have observed up to now is a simple, easy to use application with a user interface that doesn't have any bells and whistles, graphs and charts, or other unnecessary overhead that distracts the user from accomplishing their tasks. Some of the things that I am missing are supporting documentation on the import of members, the expected file format and structure to use, the simple bookkeeping of member dues and donations, or the import of bank statements for the purpose of such transactions.

Unfortunately, with other projects I am involved in next to my regular work I just don't have enough capacity to become active on a regular basis on this one. Although I do have an intermediate knowledge and development skill in Python and an understanding of Django and the MVC paradigm in general, I may be able to provide some support on the interfacing mentioned above.

It would be great to have a team of skilled developers ready and willing to take on the work of furthering the product, but at first a meeting should be held with the group who has spoken up in this thread. The topics should be the state of the product at the current version number, an overview of the functionality, a look at the enhancement requests, and possibly putting together a roadmap and setting milestones for the next release versions.

Just some thoughts from my side. I may also be able to help out with the project management, but with many software packages sometimes "it takes a village".

Thanks,

Karl

**Comment by rixx** [comment]
First off: thank you, everybody here, for responding!

@Herr-Dante is now an owner of the repo and organisation. Thank you for taking over! He mentioned he'll be rather busy for the coming month, but is happy to see the interest in maintainership and collaboration here, so please stick around / collaborate / communicate / open issues and PRs etc. Thank you all again, it's been a pleasure!

**Comment by vollkorn1982** [comment]
Ah, PGP, I've got that on my todo list since years. I have quite a bit of experience with PGP and will help anyone working on this topic.

**Comment by GoliathLabs** [comment]
> Ah, PGP, I've got that on my todo list since years. I have quite a bit of experience with PGP and will help anyone working on this topic.

I haven't had the chance to work with PGP integration yet. If @vollkorn1982 has experience with it, it would be ideal for them to take on this issue.

In the meantime, I'll focus on other tasks as soon as I can find some free time in the next few weeks.
